### PR TITLE
Add ViT-B for visual prompting

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] ViT-B for visual prompting and zero-shot visual prompting
- [x] Fix bugs
    - [x] Empty prediction in `PromptGetter` (zero-shot)
    - [x] Stack more indices than #masks in `__inspect_overlapping_areas` (zero-shot)
- [x] Improve managing `reference_info` (zero-shot)
- [x] tests
    - [x] unit test

Comparing with v1.5, fine-tuning results on wgisd dataset to check reproducibility are shown below:
version | best epoch | best mDice
-- | -- | --
v1.5 | 11 | 93.36
v2.0 | 81 | 93.33

TODOs :
- set regression test in the next PR
- update recipe
    - Performance ranges of both versions seems the same, but I'm not sure why v2.0's early stopping worked late.
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/model/module/visual_prompting.py [hunk 1]
```diff
@@ -316,8 +316,19 @@ def _inference_step(
...
+                            {
+                                "masks": torch.zeros_like(_target[0]["masks"], dtype=_target[0]["masks"].dtype),
+                                "labels": torch.zeros_like(_target[0]["labels"], dtype=_target[0]["labels"].dtype),
+                                "scores": torch.zeros(len(_target[0]["labels"]), dtype=torch.float32),
+                            }
+                            for _ in range(num_diff)
+                        ]
+                    else:
+                        for i in range(num_diff):
+                            _preds.append(_preds[i])
```
Comment (vinnamkim) [inline src/otx/core/model/module/visual_prompting.py]:
```suggestion
PAD_PREDICTION = {
                                "masks": torch.zeros_like(_target[0]["masks"], dtype=_target[0]["masks"].dtype),
                                "labels": torch.zeros_like(_target[0]["labels"], dtype=_target[0]["labels"].dtype),
                                "scores": torch.zeros(len(_target[0]["labels"]), dtype=torch.float32),
                            }

...

                    for idx in range(num_diff):
                        pred = _preds[idx] if idx < len(_preds) else PAD_PREDICTION
                        _preds.append(pred)
```

Comment (sungchul1) [reply src/otx/core/model/module/visual_prompting.py]: Done at https://github.com/openvinotoolkit/training_extensions/pull/2850/commits/a027b604d437c5310ebe0164d5036bb7fea71287.

**Review by sungmanc — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]
